lib/helper.js

```diff
@@ -139,16 +139,35 @@ function withContext(Component, context) {
  * this method is not needed if you are using babel
  *
  * @method jsxTranspile
+ * @param {Boolean} instrument the original code
  **/
-function jsxTranspile() {
+function jsxTranspile(instrument) {
     var fs = require('fs');
     var babel = require('babel');
 
-    require.extensions['.jsx'] = function (module, filename) {
-        var content = fs.readFileSync(filename, 'utf8');
-        var compiled = babel.transform(content, {nonStandard: true});
-        return module._compile(compiled.code, filename);
-    };
+    if (instrument !== undefined && instrument && instrument === 'true') {
```

Comment:
if people use jenkins-mocha then you could also read --no-coverage from argv to determine this